spec.html

```diff
@@ -11059,6 +11060,26 @@ <h1>Examples of Automatic Semicolon Insertion</h1>
       <p>In the circumstance that an assignment statement must begin with a left parenthesis, it is a good idea for the programmer to provide an explicit semicolon at the end of the preceding statement rather than to rely on automatic semicolon insertion.</p>
     </emu-clause>
   </emu-clause>
+  <emu-clause id="sec-hazards-of-automatic-semicolon-insertion">
+    <h1>Hazards of Automatic Semicolon Insertion</h1>
+    <em>This section is non-normative.</em>
+    <p>ECMAScript programs can be written in a style with very few semicolons by relying on automatic semicolon insertion. However, as described above, semicolons are not inserted at every newline, and some of these cases which do not have an automatically inserted semicolon can be counter-intuitive or confusing.</p>
```

Comment:
This paragraph is perhaps too colourful for a spec. It is perhaps the most devisive of all paragraphs in this section, and words such as `counter-intuitive` and `confusing` could constitute in some people's minds as FUD. Personally I feel the rest of the prose reads perfectly fine with _without_ this line.

Reply:
I mean, uncertainty is exactly what semicolons are designed to prevent. :disappointed:

Reply:
Counter-intuitive to my mind implies that 2 humans (a spec-lawyer and casual user) might come to different conclusions.

What if the line were rephrased to emphasize that a human reader and the parser come to different conclusions, which IIUC, is the crux of the problem.

>ECMAScript programs can be written in a style with very few semicolons
by relying on automatic semicolon insertion.
> A human reader may mentally insert semicolons by reasoning about
the author's intent, but the EmcaScript parsing algorithm uses no
such cues and may insert semicolons in different places affecting
program correctness.

That's probably too wordy, but is that the right sentiment?

Reply:
Can we answer the following two questions, confidently:

1. Does what this paragraph is saying _need to be said_?
2. Does this paragraph _add to_ or _otherwise summarise_ the remaining prose?

I'd say if the answer to the above two questions is "no" then we should remove this para. If the answer is "yes" then we should put the effort in to make sure it fulfils that goal correctly. Personally, I think the above questions answer as "no".

Reply:
sounds like, for TC39 and the future of JS, there are technical reasons, and not FUD.

Reply:
Just to be clear - whether or not _it is objectively_ FUD is irrelevant. Whether or not _it is construed to be_ FUD is the cause for concern. I feel this is demonstrable by scrolling up: the cognative dissonance is strong. Giving peoeple the opportunity to place their subjectiveness onto this prose is giving them an opportunity to dismiss it. Sticking to hard facts and less divisive language prevents this altogether.

Reply:
rephrased a bit